### PR TITLE
Added Single Video Page Please Review

### PR DESCRIPTION
**Pull request by aniketxparihar**
Components Added:
Video Actions
Pages Added
Single Video Page

### CODE REVIEW

File: src/Components/VideoActions/VideoActions.jsx [hunk 1]
```diff
@@ -2,13 +2,167 @@ import React, { useEffect, useState } from "react";
...
+  };
+
+  //Remove from Watch Later videos
+  const removeFromWatchLaterHandler = async () => {
+    const response = await axios.delete(
+      `/api/user/watchLater/${props.video._id}`,
+      {
+        headers: {
+          authorization: JSON.parse(localStorage.getItem("user")).encodedToken,
+        },
```
Comment (JowelTisso) [inline src/Components/VideoActions/VideoActions.jsx:76]: You can extract localStorage as separate function

**Review by JowelTisso — APPROVED** [review]
Looks Good

File: src/Components/VideoActions/VideoActions.jsx [hunk 2]
```diff
@@ -2,13 +2,167 @@ import React, { useEffect, useState } from "react";
 import "./VideoActions.css";
 import { useAuth } from "../../Context/Auth-Context";
 import { useUserData } from "../../Context/UserData-Context";
+import toast, { Toaster } from "react-hot-toast";
 
 import axios from "axios";
 const VideoActions = (props) => {
-  
+  const {
```
Comment (xsaurabhsuthar) [inline src/Components/VideoActions/VideoActions.jsx:9]: Suggestion: You can extract the whole logic to a separate Container named VideoActionsContainer and use it like a custom hook in the view(VideoActions) component.

File: src/Components/VideoActions/VideoActions.jsx (hunk 1 above)
Comment (xsaurabhsuthar) [reply src/Components/VideoActions/VideoActions.jsx:76]: or extract the encoded token from localStorage and store it in the state.

File: src/Pages/SingleVideo/SingleVideo.jsx [hunk 3]
```diff
@@ -1,13 +1,53 @@
-import React, { useState,useEffect } from "react";
+import React, { useState, useEffect } from "react";
 import VideoActions from "../../Components/VideoActions/VideoActions";
 import "./SingleVideo.css";
 import { useAuth } from "../../Context/Auth-Context";
 import axios from "axios";
 const SingleVideo = () => {
- 
+  const [videoData, setVideoData] = useState("");
```
Comment (xsaurabhsuthar) [inline src/Pages/SingleVideo/SingleVideo.jsx:7]: Suggestion: Similarly here also you can extract it to container.

**Review by xsaurabhsuthar — APPROVED** [review]
Need some changes suggested above otherwise, code looks good and clean. Good to merge 🚀🚀🚀🚀.

File: src/Components/VideoActions/VideoActions.jsx (hunk 1 above)
Comment (aniketxparihar) [reply src/Components/VideoActions/VideoActions.jsx:76]: I did that but it wasn't updating the state correctly for some reason so I decided to hard code it everywhere. Will check that thank you!

**Pull request merged**